parity/helpers.rs

```diff
@@ -298,6 +300,12 @@ pub fn password_prompt() -> Result<Password, String> {
 	Ok(password)
 }
 
+pub fn new_fetch_client(https_proxy: Option<&String>) -> Result<fetch::Client, String> {
+	https_proxy
```

Comment:
if you change `fetch::Client::new(https_proxy: &Option<String>)` and remove `fetch::Client::new_with_proxy(https_proxy_url: &Option<String>)`

Then you can do:

```rust
pub fn new_fetch_client(https_proxy: &Option<String>) -> Result<fetch::Client, String> {
       fetch::Client::new(https_proxy)
           .map_err(|e| format!("Error starting fetch client: {:?}", e))
 }
```

However, it requires some changes by the caller of this method ofc!